view_conversations/conv-viewer-webapp/src/app/app.component.ts

```diff
@@ -94,17 +122,65 @@ export class AppComponent implements OnInit {
     document.location.hash = JSON.stringify(objToEncode);
   }
 
+  submitCommentSearch(comment : wpconvlib.Comment) {
+    console.log('model-based form submitted');
+    this.errorMessage = null;
+    this.updateLocationHash();
+
+    this.inFlightRequest =
+        this.http
+            .get(encodeURI(
+              '/api/comment-id/' + comment.id))
+            .subscribe(
+                (actions: wpconvlib.Comment[]) => {
+                  console.log('got conversation!');
```

Comment:
This looks like quite a big bit of logic. Pull it out into its own function.

Reply:
I don't think it's a good idea, it assigns a value to comment which is an argument in the function.